packages/govuk-frontend-review/src/views/full-page-examples/cookie-banner-essential-cookies/index.mjs

```diff
@@ -1,16 +1,11 @@
-/**
- * @param {import('express').Application} app
- */
-export default (app) => {
-  app.post(
-    '/full-page-examples/cookie-banner-essential-cookies',
-    (request, response) => {
-      response.render(
-        './full-page-examples/cookie-banner-essential-cookies/index',
-        {
-          cookies: request.body.cookies
-        }
-      )
-    }
-  )
-}
+import express from 'express'
+
+const router = express.Router()
```

Comment:
**question** I didn't understand from [the commit](https://github.com/alphagov/govuk-frontend/pull/4499/commits/363b9bdd5ab3dcb2c93de455e866f58bd0474cdc)'s message what was the advantage of using a Router vs a function as:
- we could loop over a list of functions in a similar way [we loop over a list of Routers](https://github.com/alphagov/govuk-frontend/blob/6aa97229f2488bbdf748663000823301d34d4220/packages/govuk-frontend-review/src/routes/full-page-examples.mjs#L63) 
- I don't think we have unit tests (which the message mentions) around these routes
 
Could you add more details of what the move to a Router enables (or what having a function was blocking)?

Reply:
Primarily because they're small, testable, modular things and part of using Express.js: https://expressjs.com/en/guide/routing.html#express-router

But they can also have their own params, middleware and child routes too

Thanks to that [`.use('/route', router)` pattern](https://github.com/alphagov/govuk-frontend/pull/4499#discussion_r1412435460) we define paths in a single place too. When you want to change your website's URL structure, you just change the "mount point" and the attached routers come with it

Ah sorry about the unit test wording, you're right:

>We previously used wrapper functions and couldn’t easily mount routes into unit tests (like we do for middleware)

I went with a [loop of Puppeteer tests in the end](https://github.com/alphagov/govuk-frontend/pull/4499/commits/6aa97229f2488bbdf748663000823301d34d4220#diff-10d736b47114c2b27433e1f8aa229ea4ea15b24175b17c90eba49e6e12548454R18-R20)

But look at the [Banner tests in `a447f85`](https://github.com/alphagov/govuk-frontend/commit/a447f853171fa0182b53ca5758783c9a9579fae2#diff-a62640b3d51748b822da9171bfb291ba71d2a088104ad419be2fe429795b25ddR15-R25) where middleware and routers can be tested with this pattern now:

```mjs
const app = express()
const agent = supertest.agent(app)

// Add middleware
app.use(middleware.request)
app.use(middleware.banner)

// Add router
app.get('/', router)
```

Reply:
It's quite common for performance too

You can add middleware only to the routes that need it

E.g. Requiring user sign in for `/account` in a single `.use()` at router level